classes/plugins/ImportExportPlugin.inc.php

```diff
@@ -144,10 +144,20 @@ function getExportPath() {
 	 * @param $objectsFileNamePart string Part different for each object type.
 	 * @param $context Context
 	 * @param $extension string
+	 * @param $dateFilenamePart string The datetime part of the exported filename
 	 * @return string
 	 */
-	function getExportFileName($basePath, $objectsFileNamePart, $context, $extension = '.xml') {
-		return $basePath . $this->getPluginSettingsPrefix() . '-' . date('Ymd-His') .'-' . $objectsFileNamePart .'-' . $context->getId() . $extension;
+	function getExportFileName($basePath, $objectsFileNamePart, $context, $extension = '.xml', $dateFilenamePart = null, $dateFilenamePartFormat = 'Ymd-His') {
```

Comment:
@defstat, I think this is going to break several functions that make use of `getExportFileName` in different import/export plugins; can you try making these improvements without modifying the function signature?

Reply:
@asmecher I can easily remove the second additional parameter but I need to know the `$dateFilenamePart` of the filename, so that I can pass it to the form. I could not think a better way to do that, than creating it outside the `getExportFileName`  and passing it as a parameter. All the functions that are using it already, will use the default value, which is null, so the code will again create the same date part as before. 

I could, probably, extract the part of the filename that corresponds to the datetime (for example take the string between the first `-` character and the third `-` character) - but then I have the problem of recreating the filename path (without copying the `getExportFileName` code to the plugin)

Adding the name in the parameters list serves both purposes.

If you have another idea for that I can try it.